### ISSUE TITLE
Testlib: If skipping a test, the setup should also skip

### ISSUE THREAD

**Issue by castulo** [clearlinux/swupd-client#1486]
Currently when a test is marked to be skipped, the test_setup for that test is still run. This could be skipped since the test is not going to run.

Currently we are considering caching the creation of test resources, if we do that then this issue is no longer relevant.

**Comment by otaviobp** [comment]
It would be great to be able to set all tests of a file to be skipped only once instead of having multiple checks for RUNNING_IN_CI like we have today
